### PR TITLE
mm-te: refactor helm chart to deploy MM-TE using the configuration in the DB

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
This PR fixes the problem with the config.json being read-only, after mattermost 5.30+ looks like it tries to save some data in the config.json, to avoid that we can use the configuration in the DB and also the environment variables.

Before upgrading an existing installation, please check the readme for the steps in how to migrate to the newer chart

It was tested with a deployment using the latest helm chart, then applied the migration steps and upgrade to the chart in this PR.

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost-helm/issues/100

### CODE REVIEW

**Review by metanerd — APPROVED** [review]
LGTM! :tada:

**Review by stylianosrigas — APPROVED** [review]
Great stuff @cpanato 🚀

File: charts/mattermost-team-edition/README.md [hunk 1]
```diff
@@ -46,10 +46,69 @@ provide you with a custom message on what you need to change in your
...
 made to the k8s cluster.
 
-## Upgrading  the Chart to 4.0.0+
+## Upgrading the Chart to 4.0.0+
 
 The Chart version 4.0.0+ supports only Helm v3, follow the [guide](https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/)
 
+## Upgrading the Chart to 5.0.0+
+
+There are some changes that was introduced in Mattermost version 5.30+ that made this helm chart incompatible and the upgrade to fail. To fix that a new Major chart was relased to fix that, and now the configuration is moved to the Database and to environment variables
```
Comment (jwilander) [inline charts/mattermost-team-edition/README.md:55]:
```suggestion
There were some changes that was introduced in Mattermost version 5.30+ that made this helm chart incompatible and caused the upgrade to fail. To fix that a new Major version of the chart was released, and now the configuration is moved to the database and environment variables.
```

File: charts/mattermost-team-edition/README.md [hunk 2]
```diff
@@ -46,10 +46,69 @@ provide you with a custom message on what you need to change in your
...
+{"level":"info","msg":"Successfully migrated config."}
+```
+
+5 - Update the helm repo
+
+```
+$ helm repo update
+```
+
+6 - upgrade your mattermost helm, check your existing values, you remove the section `config.json` from your custom values
```
Comment (jwilander) [inline charts/mattermost-team-edition/README.md:94]:
```suggestion
6 - Upgrade your mattermost helm, check your existing values, you remove the section `config.json` from your custom values
```

File: charts/mattermost-team-edition/README.md [hunk 3]
```diff
@@ -46,10 +46,69 @@ provide you with a custom message on what you need to change in your
...
+
+7 - It should deploy and in a few seconds the new Pod for the Mattermost server should be up and running
+
+8 - After all up and running you can upgrade it again to get the latest image tag
+
+```
+$ helm upgrade  mm-te -f custom_values.yaml mattermost/mattermost-team-edition
+```
+
+if in your `custom_values.yaml` you set the image.tag, please update that to the latest available, at the time of this doc was write the version is `5.35.3`
```
Comment (jwilander) [inline charts/mattermost-team-edition/README.md:109]:
```suggestion
If in your `custom_values.yaml` you set the image.tag, please update that to the latest available, at the time of this doc was write the version is `5.35.3`
```

**Review by jwilander — APPROVED** [review]
Good stuff 👍 Just a few grammar suggestions

**Pull request merged**